### DISCUSSION TITLE
Complex Table Extraction from PDF

### DISCUSSION THREAD

**Discussion by gayathri-pan** [pymupdf/PyMuPDF-Utilities#137] [Q&A]
hello,

i am trying to extract tables from the pdf using table-analysis/find_tables.ipynb and my pdf have some complex tables. kindly guide me how to do it instead doing manually for each table.

**Comment by JorjMcKie** [comment]
Please be more specific with what you need - provide examples, etc.

**Comment by gayathri-pan** [comment]
[stm32f103c8 (1).pdf](https://github.com/user-attachments/files/15567645/stm32f103c8.1.pdf)
the above pdf have lot of merged and split tables. 

kindly guide me how to extract  all kind of table

Thanks in advance!

**Reply by JorjMcKie** [reply]
Did you read the documentation of [find_tables](https://pymupdf.readthedocs.io/en/latest/page.html#Page.find_tables)? Please start with extracting tables using this method and ask specific questions when needed.
